x-pack/plugins/canvas/public/components/dom_preview/dom_preview.js

```diff
@@ -12,39 +12,46 @@ export class DomPreview extends React.Component {
   static container = null;
   static content = null;
   static observer = null;
+  static original = null;
+  static updateTimeout = null;
 
   static propTypes = {
     elementId: PropTypes.string.isRequired,
     height: PropTypes.number.isRequired,
   };
 
   componentDidMount() {
-    const original = document.querySelector(`#${this.props.elementId}`);
-
-    const update = this.update(original);
-    update();
-
-    const slowUpdate = debounce(update, 250);
-
-    this.observer = new MutationObserver(slowUpdate);
-    // configuration of the observer
-    const config = { attributes: true, childList: true, subtree: true };
-    // pass in the target node, as well as the observer options
-    this.observer.observe(original, config);
+    this.update();
   }
 
   componentWillUnmount() {
-    this.observer.disconnect();
+    clearTimeout(this.updateTimeout);
+    this.observer && this.observer.disconnect(); // observer not guaranteed to exist
   }
 
-  update = original => () => {
+  update = () => {
     if (!this.content || !this.container) {
       return;
     }
 
-    const thumb = original.cloneNode(true);
+    if (!this.observer) {
+      this.original = this.original || document.querySelector(`#${this.props.elementId}`);
+      if (this.original) {
+        const slowUpdate = debounce(this.update, 100);
+        this.observer = new MutationObserver(slowUpdate);
+        // configuration of the observer
+        const config = { attributes: true, childList: true, subtree: true };
+        // pass in the target node, as well as the observer options
+        this.observer.observe(this.original, config);
+      }
+      clearTimeout(this.updateTimeout); // to avoid the assumption that we fully control when `update` is called
```

Comment:
Q/CR: Shouldn't this be in an else? If `this.original` exists in time, is there a point in calling `this.update` again (which is all this bit of code does AFAICT)?

```diff
+      } else {
+        // keep trying if original is missing...
+        // to avoid the assumption that we fully control when `update` is called
+        clearTimeout(this.updateTimeout);
+        this.updateTimeout = setTimeout(this.update, 30);
+        return;
       }
-      clearTimeout(this.updateTimeout); // to avoid the assumption that we fully control when `update` is called
-      this.updateTimeout = setTimeout(this.update, 30);
-      return;
```

Reply:
Thanks, made the change, btw. this ` ```diff` in the commit is neat!